plugins/basic/nexus-repository-raw/src/main/java/org/sonatype/nexus/repository/raw/internal/RawContentFacetImpl.java

```diff
@@ -217,9 +218,12 @@ public boolean delete(final String path) throws IOException {
       if (component == null) {
         return false;
       }
+
+      final SearchFacet searchFacet = getRepository().facet(SearchFacet.class);
+      final SearchItemId searchId = searchFacet.identifier(component);
```

Comment:
BTW, OrientDB does support the notion of hooks: http://orientdb.com/docs/last/Hook.html but these are more low-level